core/unit_test/CMakeLists.txt

```diff
@@ -395,50 +413,49 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
+  KOKKOS_OPTION( ${DEVICE}_EXCLUDE_TESTS "" STRING "Incremental test exclude list" )
   # Add unit test main
-  SET(${BACK_END_NAME}_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/UnitTestMainInit.cpp)
+  SET(${DEVICE}_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/UnitTestMainInit.cpp)
 
   # Iterate over incremental tests in directory
 
   APPEND_GLOB(INCREMENTAL_FILE_LIST ${CMAKE_CURRENT_SOURCE_DIR}/incremental/*.hpp)
-  
+
+  SET(DEVICE_NAME ${KOKKOS_${DEVICE}_NAME})
   FOREACH (CURRENT_FILE_PATH ${INCREMENTAL_FILE_LIST})
     GET_FILENAME_COMPONENT( CURRENT_FILE_NAME ${CURRENT_FILE_PATH} NAME )
     STRING (REPLACE ".hpp" "" CURRENT_TEST_NAME ${CURRENT_FILE_NAME})
-    IF (NOT CURRENT_TEST_NAME IN_LIST Kokkos_${BACK_END_NAME}_EXCLUDE_TESTS)
-       SET (CURRENT_TEST_OUTPUT_FILENAME ${CURRENT_TEST_NAME}_${BACK_END_NAME})
-       FILE( STRINGS ${CURRENT_FILE_PATH} CURRENT_REQUIRED_FEATURE_LINE REGEX "Kokkos_Feature_Level_Required" ) 
+    IF (NOT CURRENT_TEST_NAME IN_LIST Kokkos_${DEVICE}_EXCLUDE_TESTS)
+       SET (CURRENT_TEST_OUTPUT_FILENAME ${CURRENT_TEST_NAME}_${DEVICE})
+       FILE( STRINGS ${CURRENT_FILE_PATH} CURRENT_REQUIRED_FEATURE_LINE REGEX "Kokkos_Feature_Level_Required" )
        # From each test get level implementation required
        STRING( REGEX REPLACE ".*Kokkos_Feature_Level_Required:" "" CURRENT_REQUIRED_FEATURE_LEVEL ${CURRENT_REQUIRED_FEATURE_LINE} )
        # Cross-reference list of dependencies with selected feature list > matching feature test files are added to test applications
-       IF (Kokkos_${BACK_END_NAME}_FEATURE_LEVEL GREATER_EQUAL CURRENT_REQUIRED_FEATURE_LEVEL) 
+       IF (KOKKOS_${DEVICE}_FEATURE_LEVEL GREATER_EQUAL CURRENT_REQUIRED_FEATURE_LEVEL)
           CONFIGURE_FILE (IncrementalTest.cpp.in ${CMAKE_BINARY_DIR}/core/unit_test/generated/${CURRENT_TEST_OUTPUT_FILENAME}.cpp )
-          SET(${BACK_END_NAME}_SOURCES ${${BACK_END_NAME}_SOURCES}; ${CMAKE_BINARY_DIR}/core/unit_test/generated/${CURRENT_TEST_OUTPUT_FILENAME}.cpp)
+          SET(${DEVICE}_SOURCES ${${DEVICE}_SOURCES}; ${CMAKE_BINARY_DIR}/core/unit_test/generated/${CURRENT_TEST_OUTPUT_FILENAME}.cpp)
        ENDIF()
      ENDIF()
   ENDFOREACH()
 
-  STRING(TOUPPER ${BACK_END_NAME} UC_BACK_END_NAME)
+  STRING(TOUPPER ${DEVICE} UC_DEVICE)
 
   KOKKOS_OPTION (
-    ENABLE_${UC_BACK_END_NAME} ON BOOLEAN "ENABLE ${UC_BACK_END_NAME}"
+    ENABLE_${UC_DEVICE} ON BOOLEAN "ENABLE ${UC_DEVICE}"
   )
 
   KOKKOS_ADD_EXECUTABLE_AND_TEST(
-    IncrementalTest_${BACK_END_NAME}
-    SOURCES ${${BACK_END_NAME}_SOURCES}
+    IncrementalTest_${DEVICE}
+    SOURCES ${${DEVICE}_SOURCES}
   )
 
-  TARGET_INCLUDE_DIRECTORIES( ${PACKAGE_NAME}_IncrementalTest_${BACK_END_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/incremental )
+  TARGET_INCLUDE_DIRECTORIES( ${PACKAGE_NAME}_IncrementalTest_${DEVICE} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/incremental )
 
 ENDFUNCTION()
 
-KOKKOS_OPTION(INCREMENTAL_TEST_BACK_ENDS "" STRING "List of backends to create incremental tests for")
```

Comment:
So you want to build and run the incremental tests for all enabled devices unconditionally?

Reply:
yes